src/skmultiflow/lazy/knn_classifier.py

```diff
@@ -1,49 +1,54 @@
     warnings.warn("'KNN' has been renamed to 'KNNClassifier' in v0.5.0.\n"
                   "The old name will be removed in v0.7.0", category=FutureWarning)
     return KNNClassifier(n_neighbors=n_neighbors,
                          max_window_size=max_window_size,
-                         leaf_size=leaf_size,
-                         nominal_attributes=nominal_attributes)
+                         leaf_size=leaf_size)
 
 
-class KNNClassifier(BaseSKMObject, ClassifierMixin):
+class KNNClassifier(BaseNeighbors, ClassifierMixin):
     """ k-Nearest Neighbors classifier.
     
-    This non-parametric classification method keeps a data window with the last max_window_size
-    training samples. The predicted class-label for a given query sample is obtained in two steps:
-    first, find the closest n_neighbors to the query sample in the data window. Second, aggregate
-    the class-labels of the n_neighbors to define the predicted class for the query sample.
+    This non-parametric classification method keeps track of the last
+    ``max_window_size`` training samples. The predicted class-label for a
+    given query sample is obtained in two steps:
+
+    1. Find the closest n_neighbors to the query sample in the data window.
+    2. Aggregate the class-labels of the n_neighbors to define the predicted
+       class for the query sample.
 
-    
     Parameters
     ----------
     n_neighbors: int (default=5)
         The number of nearest neighbors to search for.
         
     max_window_size: int (default=1000)
-        The maximum size of the window storing the last viewed samples.
+        The maximum size of the window storing the last observed samples.
         
     leaf_size: int (default=30)
-        The maximum number of samples that can be stored in one leaf node, 
-        which determines from which point the algorithm will switch for a 
-        brute-force approach. The bigger this number the faster the tree 
-        construction time, but the slower the query time will be.
-        
-    nominal_attributes: numpy.ndarray (optional, default=None)
-        List of Nominal attributes. If empty, then assume that all attributes are numerical.
+        sklearn.KDTree parameter. The maximum number of samples that can
+        be stored in one leaf node, which determines from which point the
+        algorithm will switch for a brute-force approach. The bigger this
+        number the faster the tree construction time, but the slower the
+        query time will be.
+
+    metric: string or sklearn.DistanceMetric object
+        sklearn.KDTree parameter. The distance metric to use for the KDTree.
+        Default=’euclidean’. KNNClassifier.valid_metrics() gives a list of
+        the metrics which are valid for KDTree.
     
     Notes
     -----
-    This classifier is not optimal for a mixture of categorical and numerical features.
+    This estimator is not optimal for a mixture of categorical and numerical
```

Comment:
Should we explicit that categorical features are going to be treated as numeric?

Reply:
Updated in 31687ca4743f3fa03f752ed182135cbef9fc7901